### PR TITLE
allow project to be built with clang on Linux

### PR DESCRIPTION
**Pull request by sethalves**
- allow project to be built with clang on Linux

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- no functional change
- build on Linux after doing ```
export CC=/usr/bin/clang
export CXX=/usr/bin/clang++
```

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3960/HighFidelity-Beta-PR11549-4d2f16efc3f1d37521f761c30d4c513a4b9f06b1.exe

File: assignment-client/src/audio/AudioMixerSlavePool.cpp [hunk 1]
```diff
@@ -97,7 +97,11 @@ void AudioMixerSlavePool::run(ConstIter begin, ConstIter end) {
 #else
     // fill the queue
     std::for_each(_begin, _end, [&](const SharedNodePointer& node) {
+#if defined(__clang__) && defined(Q_OS_LINUX)
+        _queue.push(node);
+#else
         _queue.emplace(node);
+#endif
```
Comment (hyperlogic) [inline assignment-client/src/audio/AudioMixerSlavePool.cpp]: _queue is a tbb::concurrent_queue<T> perhaps you do not have the right version of tbb?

File: libraries/networking/src/LimitedNodeList.cpp [hunk 2]
```diff
@@ -676,7 +676,7 @@ SharedNodePointer LimitedNodeList::addOrUpdateNode(const QUuid& uuid, NodeType_t
         }
 
         // insert the new node and release our read lock
-#ifdef Q_OS_ANDROID
+#if defined(Q_OS_ANDROID) || (defined(__clang__) && defined(Q_OS_LINUX))
         _nodeHash.insert(UUIDNodePair(newNode->getUUID(), newNodePointer));
 #else
         _nodeHash.emplace(newNode->getUUID(), newNodePointer);
```
Comment (hyperlogic) [inline libraries/networking/src/LimitedNodeList.cpp]: This also might be tbb related.
    
    typedef tbb::concurrent_unordered_map<QUuid, SharedNodePointer, UUIDHasher> NodeHash;

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3992/HighFidelity-Beta-PR11549-3cef3779df7c2376af7a9fd43d89861f8d216b69.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3992/HighFidelity-Beta-PR11549-3cef3779df7c2376af7a9fd43d89861f8d216b69.exe

**Comment by OmegaHeron** [comment]
Tested Ubuntu 16.04, Ubuntu 17.10 and Debian 10
QA Pass

Compiled on all without flag set.  Compiled on all with flag set.  In both cases compile went as expected and produced viable binaries.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4055/HighFidelity-Beta-PR11549-ca128a3251aec7073854ccbe3d4c6d0c55c2467d.dmg

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4059/HighFidelity-Beta-PR11549-ca128a3251aec7073854ccbe3d4c6d0c55c2467d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4060/HighFidelity-Beta-PR11549-ca128a3251aec7073854ccbe3d4c6d0c55c2467d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4060/HighFidelity-Beta-PR11549-ca128a3251aec7073854ccbe3d4c6d0c55c2467d.exe

**Pull request merged**